### PR TITLE
Revamp cltbld user creation process for macOS 13

### PR DESCRIPTION
**Pull request by rcurranmoz**
Problem: `cltbld` user account creation on macOS 13+ does not land the correct entitlements for `LaunchAgents` to function as they should.

While this PR converges fine there is still a few more commits that need to take place, specifically around the `require` bits i commented out to get it working AND ` macos_utils::clean_appstate`

The reason I'm cutting a PR this early  is that I really need a sanity check from @aerickson and `pre-commit` is unhappy for some reason.

### CODE REVIEW

**Comment by rcurranmoz** [comment]
Mostly working. Need to suppress the Accessibility window at first login

**Comment by rcurranmoz** [comment]
Deployed a config profile to suppress Accessibility dialog

File: modules/macos_utils/manifests/clean_appstate2.pp [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+define macos_utils::clean_appstate2 (
```
Comment (aerickson) [inline modules/macos_utils/manifests/clean_appstate2.pp:5]: Can we come up with a better name for this class?

File: modules/roles_profiles/manifests/profiles/cltbld_user.pp [hunk 2]
```diff
@@ -3,107 +3,132 @@
...
-        'Darwin': {
-            $password     = lookup('cltbld_user.password')
-            $salt         = lookup('cltbld_user.salt')
-            $iterations   = lookup('cltbld_user.iterations')
-            $kcpassword   = lookup('cltbld_user.kcpassword')
+  case $facts['os']['name'] {
+    'Darwin': {
+      $account_username = 'cltbld'
+      $password     = lookup('cltbld_user.password')
+      $password2    = 'cltbld'
```
[the remark this answers is not in the document]
Comment (rcurranmoz) [reply modules/roles_profiles/manifests/profiles/cltbld_user.pp:10]: Renamed to `clean_appstate_13_plus`

File: modules/macos_utils/manifests/clean_appstate2.pp (hunk 1 above)
Comment (rcurranmoz) [reply modules/macos_utils/manifests/clean_appstate2.pp:5]: Renamed to `clean_appstate_13_plus`

File: modules/roles_profiles/manifests/profiles/cltbld_user.pp [hunk 3]
```diff
@@ -7,7 +7,7 @@
     'Darwin': {
       $account_username = 'cltbld'
       $password     = lookup('cltbld_user.password')
-      $password2    = 'cltbld'
+      $password_unhashed    = lookup('cltbld_user.unhasedpassword')
```
Comment (aerickson) [inline modules/roles_profiles/manifests/profiles/cltbld_user.pp:10]: typo? `cltbld_user.unhasedpassword`

**Review by aerickson — APPROVED** [review]
lgtm. ship it!

**Pull request merged**